### ISSUE TITLE
How to change properties after init

### ISSUE THREAD

**Issue by chrys1110** [Foliotek/Croppie#22]
I first init a croppie by
```var basic = $cropper.croppie({
    viewport: {
        width: 200,
        height: 100
    },
    boundary: {
        width: 250,
        height: 150
    },
});```
then bind it to an image.
When binding it to a different one later how can I change the viewport and boundary afterwards?

**Comment by chrys1110** [comment]
Or if this is impossible:
How could I destroy and reinitialize it?

**Comment by thedustinsmith** [comment]
I've just added a destroy method and it will be included in the next release of croppie.

I like the idea of being able to change properties, I'm just not sure where it fits into the API, yet.

I'll keep this issue open until we figure that out.  Thanks for the suggestion!

**Comment by chrys1110** [comment]
Thanks for your quick response! Do you think you could push the next release anytime soon? Only then I can go forward with my project.

**Comment by thedustinsmith** [comment]
I released it later yesterday - the destroy method is in `1.0.2`

https://github.com/Foliotek/Croppie/releases/tag/v1.0.2

Let me know if it doesn't work for you.

**Comment by chrys1110** [comment]
Great! It works fine. Thanks for your effort!

**Comment by chrys1110** [comment]
And for the properties it might be a good way to set the properties like you do in the the "bind" method.

**Comment by thedustinsmith** [comment]
I agree that the bind method would be a good place to set these properties.  I just recently added a `refesh()` method for when the cropper was not initially visible.  I'm wondering if it would be better replaced by an empty `bind()`.  

Again, `1.0.3` - thanks for identifying these items.  Super helpful!

**Comment by thedustinsmith** [comment]
I decided not to put this one in `1.0.3` ... doesn't seem worth the work right now.  I'd be open to a PR if you really want this functionality.

**Comment by gdoron** [comment]
I need it too.

**Comment by garek007** [comment]
Have you added this yet?

**Comment by thedustinsmith** [comment]
@garek007 Nope

**Comment by garek007** [comment]
Would love to help via a PR if I can. Can you give me a quick rundown of how and where to modify your code to make it work? Or is it not that simple?

**Comment by thedustinsmith** [comment]
I'd love to.  My explanation is going to focus around the boundary and the viewport elements, since I'm assuming those are the properties most people are wanting to modify.

So we create all the elements necessary for croppie in the `_create` function.  There's already some code in there that set the width and height of the boundary and viewport.  Those can probably be reused in a separate function.

After resizing those elements appropriately, you'll have to call `_updatePropertiesFromImage`, which won't do anything because the croppie will think it's already bound (see `self.data.bound`).  So I'm guessing the changing of properties will also have to modify that value.

After those changes, you're kind of on your own.  I'm available for questions if you run into problems.  But that's all the further I've thought this through.

Beware - changes might conflict with other changes being made for this case: #171.

Hope that helps.

**Comment by garek007** [comment]
Yes. This might be out of my skill level, but I'll take a look and see what I can come up with. I was able to reset the viewbox boundary outside of your code, but when I saved the image it did not reflect the updated boundary. I'll see what I can come up with. Thanks!

**Comment by garek007** [comment]
I just realized that what I was doing yesterday was actually working, so I don't think I'll need to modify your code. I am using this code to reset the viewport after I crop my first image.
$('.cr-viewport').css({
	'height': 50,
	'width': 50
});
$(".cr-slider").attr("max",.6);
$(".cr-slider").attr("min",0);

 I'll need to find a relationship between the viewport size and the max and min values, but this rough draft is working. 

You can see my full code here:
http://www.riptide7.com/mylittleimagecropper.com/buildemail2.php

Once I get it cleaned up I'll post it on my GitHub with Croppie as a dependency (assuming I get it working 100% right)

**Comment by JimCook57** [comment]
+1  Need this too!   To be able to change the viewport after croppie has been initialized.

**Comment by eduxavier** [comment]
+1 
Yeah! I need this too!

**Comment by cromartie1984** [comment]
+1

**Comment by LiquidSnake2ITA** [comment]
+1

**Comment by AttilaBre** [comment]
+1

**Comment by mrzarkovic** [comment]
+1

**Comment by vojtys** [comment]
+1

**Comment by bunnyhu** [comment]
Hi, I just start to use and integrate this one to my admin site. It would be nice to see some feature, I am list it then you can choice you would be like it or not.

1. Change options after initialization. A setter/getter method for options. For example bound the enableResize to a checkbox, or change the crop size from input box will be useful.
2. Get information from the uploaded image. I am not a JS pro, but maybe we know the original size, perhaps the file name, exif info, etc. Just give back an array with all info. It would be nice to make the UX better and show information during the editing.
3. A far more detailed documentation. With example for the methods. Details for the object that I get back from get() / update() ... I must check it with console.log because there is simple no information. ( array[orientation int, points array[4], zoom float] )

I hope I can give you some new idea. I am sorry I am not a js pro (I working on server side languages, I am just beginner on js) so I cannot write it for you.

**Comment by Jtmw11** [comment]
Bump

**Comment by golobitch** [comment]
Another bump

**Comment by haleybristol** [comment]
babump

**Comment by dynasource** [comment]
+1

**Comment by tednyberg** [comment]
➕ 1️⃣ 

👊 (fistbump)

**Comment by Koban2dot0** [comment]
bump

**Comment by Danishrajwizardinnovation** [comment]
Can we pass the data image in the url like the code below ? 

image_crop.croppie('bind', { 
             url:  'data:image/jpeg;base64,/9j/'
 });

**Comment by theodorejb** [comment]
I developed a fork of Croppie which includes a `setOptions` method for dynamically changing options on an existing instance. You can see this in action on the demo site, where dragging the viewport width/height sliders updates the settings in real time: https://devtheorem.github.io/cropt/
